task02/script.js

```diff
@@ -0,0 +1,109 @@
+    });
+
+    convertFrom.addEventListener("change", (e) => { //event listener added for change event of the convert from selection field
+        convertToHolder.setAttribute("class", "hidden"); //setting the div to hidden
+        convertTo.innerHTML = ""; //resetting the innerHTML of the selection options to keep from repeating
+        switch(convertFrom.value){ //switch statement based on value chosen in the convertFrom select box
+            case "Meters": //if meters is chosen
+                    createSelectOption("feet", "Feet"); //creating select option for feet 
+                    createSelectOption("miles", "Miles"); //creating select option for miles 
+                    convertToHolder.setAttribute("class", "visible"); //setting the div back to visible
+                    convertLink.addEventListener("click", (e) => { //adding event listener for a click on the convert anchor
+                        resultsHolder.innerHTML = ""; //resetting value of results holder div to blank
+                        if(convertTo.value === "Feet"){ //if feet value is selected
+                            let feet = input.value * 3.2808; //calculating the input and storing new value into variable
+                            resultsHolder.innerHTML = feet + " Feet"; //displaying the calculation
+                        } else if(convertTo.value === "Miles"){ //if miles value is selected
+                            let miles = input.value * 0.00062137; //calculating the input and storing new value into variable
+                            resultsHolder.innerHTML = miles + " Miles"; //displaying the calculation
+                        }
+                    });
+                    break; //break out of switch
+            case "Liters":
+                    createSelectOption("gallons", "Gallons"); //creating select option for gallons
+                    createSelectOption("quarts", "Quarts"); //creating select option for quarts
+                    convertToHolder.setAttribute("class", "visible");  //setting the div back to visible
+                    convertLink.addEventListener("click", (e) => { //adding event listener for a click on the convert anchor
+                        resultsHolder.innerHTML = ""; //resetting value of results holder div to blank
+                            if(convertTo.value === "Gallons"){ //if user selects gallons
+                                let gallons = input.value * 0.26417; //calculating the input and storing new value into variable
+                                resultsHolder.innerHTML = gallons + " Gallons"; //displaying the calculation
+                            } else if(convertTo.value === "Quarts"){ //if user selects quarts
+                                let quarts = input.value * 1.056688; //calculating the input and storing new value into variable
+                                resultsHolder.innerHTML = quarts + " Quarts"; //displaying the calculation
+                            }
```

Comment:
interesting to embed these listeners inside the switch. it works - it just is spread out. If you did want a single listener for the convertLink - how would that look? I'd anticipate another switch....just something to think about. Nice job!